### ISSUE TITLE
beam + note@stem.dir interaction

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#450]
The directions of beams are defined by the `@stem.dir` of the last note in the beam.  Here is a demonstration:

<img width="789" alt="screen shot 2017-01-21 at 11 04 15" src="https://cloud.githubusercontent.com/assets/3487289/22176953/c464b422-dfc9-11e6-8d96-abc2625b2994.png">

Notes marked with a "d" posses `@stem.dir="down"` in the MEI data below.  Only when the attribute is given from the last note in the beam does the beam change direction.

It would seem more logical to do one of these instead:
* take the direction from the first note of the beam
* take the direction from the first note which has an explicit `@stem.dir`.
* add an attribute `beam@dir="above|below"` which controls the placement of the beams and hence the stem directions of the notes attached to it, with `note@stem.dir` ignored for notes within a `beam` group.  This seems like the most correct thing to do.

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2017-01-21 11:00:21</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.14-dev-ea2b196</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000111948200">
                        <staffGrp xml:id="m-000000126868489">
                            <staffDef xml:id="staffdef-000000042193281" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000014356475">
                        <measure xml:id="measure-L2" n="1" right="end">
                            <staff xml:id="staff-L2F1" n="1">
                                <layer xml:id="layer-L2F1" n="1">
                                    <beam xml:id="beam-000000116637527">
                                        <note xml:id="note-L3F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L4F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L5F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L6F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000124634882">
                                        <note xml:id="note-L7F1" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="down" />
                                        <note xml:id="note-L8F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L9F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L10F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000102697413">
                                        <note xml:id="note-L11F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L12F1" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="down" />
                                        <note xml:id="note-L13F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L14F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000070967754">
                                        <note xml:id="note-L15F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L16F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L17F1" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="down" />
                                        <note xml:id="note-L18F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000050287486">
                                        <note xml:id="note-L19F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L20F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L21F1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L22F1" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="down" />
                                    </beam>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
same as #414

**Comment by lpugin** [comment]
Fixed with 7de376debeaa69323358c03f053b1b24595ab0b2

**Comment by craigsapp** [comment]
Looks good.  What is the new algorithm?  It seems to be:

(1) Search for stem directions on all notes.  If all of the explicit directions match, then use that direction.
(2) If there are contrary directions on stems then automatically decide the direction.
(3) If there are no directions, then automatically decide the direction.

Is there a stem direction attribute for beams (or will that be implemented after MEI 4.0 comes out)?

<img width="888" alt="screen shot 2017-03-06 at 1 16 31 pm" src="https://cloud.githubusercontent.com/assets/3487289/23630277/63c5381c-026f-11e7-8dc7-69491e4a004f.png">
